NLSY/nlsy_imputation.qmd

```diff
@@ -0,0 +1,227 @@
+           # Also, it seems that this represents the wealth of the household in which a respondent lives. That means that for a respondent who lives with only one parent, the net-worth variable represents the wealth of that parent.
+           # In the case of an independent respondent, contains respondent's, rather than parents', wealth.
+    ) %>%
+    mutate(mom_age = mom_age_birth + age_resp)
+
+# Making dummy or factor variables
+test <- test |>
+    mutate(
+        mom_educ_hs = factor(nlsy_encode_educ4(mom_education)),
+        dad_educ_hs = factor(nlsy_encode_educ4(dad_education)),
+        race = factor(race),
+        hisp = factor(hispanic),
+        race_eth = ifelse(hisp == "Yes" & race == "White", "White Hispanic",  NA), # Race and ethnicity interaction
+        race_eth = ifelse(hisp == "No" & race == "White", "White NonHispanic", race_eth),
+        race_eth = ifelse(hisp == "Yes" & race == "Black or African American", "Black Hispanic", race_eth),
+        race_eth = ifelse(hisp == "No" & race == "Black or African American", "Black NonHispanic", race_eth),
+        race_eth = ifelse(hisp == "Yes" & race == "American Indian, Eskimo, or Aleut", "NonBW Hispanic", race_eth),
+        race_eth = ifelse(hisp == "Yes" & race == "Asian or Pacific Islander", "NonBW Hispanic", race_eth),
+        race_eth = ifelse(hisp == "Yes" & race == "Something else? (SPECIFY)", "NonBW Hispanic", race_eth),
+        race_eth = ifelse(hisp == "No" & race == "American Indian, Eskimo, or Aleut", "NonBW NonHispanic", race_eth),
+        race_eth = ifelse(hisp == "No" & race == "Asian or Pacific Islander", "NonBW NonHispanic", race_eth),
+        race_eth = ifelse(hisp == "No" & race == "Something else? (SPECIFY)", "NonBW NonHispanic", race_eth),
+        race_eth = factor(race_eth))
+
+test$ret_sav <- ifelse(test$savings == "YES", 1, 0)
+test$hown <- ifelse(test$home == "OWNS OR IS BUYING; LAND CONTRACT", 1, 0)
+test$both_pars <- ifelse(test$both_parents == "Yes", 1, 0)
+test$par_dec <- ifelse(test$par1_deceased == "Yes" | test$par2_deceased == "Yes", 1, 0) # Either parent deceased
+
+```
+
+```{r}
+# Looking for significance with non-missing income and wealth
+non_missing <- test %>%
+    filter(!is.na(pincome),
+           !is.na(pnetworth))
+
+summary(model <- lm(pincome ~ mom_age + hisp + race + mom_educ_hs + dad_educ_hs + ret_sav + hown + both_pars + par_dec, data = non_missing))
+#Non-significant: par_dec - dropping for now
+```
+
+### Checking for multi-collinearity - correlation plot for numeric variables
+
+```{r}
+
+testing_cor <- test %>%
+    select(-id, -hispanic, -race, -mom_education, -dad_education, -savings, -home, -both_parents, -par1_deceased, -par2_deceased, - mom_age_birth, -pincome, -pnetworth, -par_dec, -mom_educ_hs,-dad_educ_hs, -hisp,-race_eth) %>%
+    drop_na()
+
+M <- cor(testing_cor)
+corrplot(M, type = "upper")
+```
+
+### Checking for multi-collinearity - VIF and Successive addition of regressors
+
```

Comment:
Some notes here would be helpful, especially with respect to vif, which I am not familiar with.

Reply:
- The documentation has more technical information about VIF: https://rdrr.io/cran/car/man/vif.html.
- The Wiki has the formula and steps for calculation: https://en.wikipedia.org/wiki/Variance_inflation_factor. The intuition is that you regress a given variable against all other regressors and calculate the coefficient of determination (1/(1-R^2)).
- For practical purposes, a rule of thumb is that VIF values above 10 are considered indications of multicollinearity, and it looks like in this code chunk, none of the values come close to that, which is good.

Reply:
Thanks Judah!